### DISCUSSION TITLE
Which Shell?

### DISCUSSION THREAD

**Discussion by tbmuc** [ImageMagick/ImageMagick#2441] [Help] [answered]
Hello I would like to create round edges for existing pictures and found this script

convert lena.jpg \
\( -size 25x25 xc:black -fill white \
	-draw "circle 25,25 25,0" -alpha off \
	-write mpr:corner +delete \) \
\( -clone 0 -fill white -colorize 100% \
	\( mpr:corner \) -gravity northwest -compose over -composite \
	\( mpr:corner -rotate 90 \) -gravity northeast -compose over -composite \
	\( mpr:corner -rotate 180 \) -gravity southeast -compose over -composite \
	\( mpr:corner -rotate 270 \) -gravity southwest -compose over -composite \) \
-alpha off -compose copy_opacity -composite lena_corners25c.png

here
http://www.imagemagick.org/discourse-server/viewtopic.php?f=1&t=30365&sid=c8f9b5a7a01d0284dfa3f85fbe6dd085&start=15 .

Which shell language is this? Normally I use IM 7 with bat file, but in this case I tried windows bat and the bash shell for windows and got many errors.

**Comment by fmw42** [comment]
It appears to be Windows CMD code. But it is missing all the end of line escapes ^. If you want to run it as is, just make it one long line of code by removing the new line markers at the end of each line in a text editor so that you get just one long line with spaces where the end of lines are located. Otherwise, in Windows add ^ to the end of each line with no spaces after the ^. Then paste that multi-line code into your CMD windows to run it. If you need to convert to .bat. You must double the % to %%.

**Comment by GeeMack** [comment, marked as answer]
It looks like the original command had many backslashes "\\" that disappeared when you pasted it in here. It would probably work in Windows CMD if you deleted all those backslashes and put a caret "^" at the end of each line as fmw42 explained above.

But since you're running ImageMagick v7, there might be much more direct ways to get rounded corners. Here's an example command that would clip each corner with a 20 pixel radius and leave it transparent in the background.

    magick image.png -alpha set -background none -size %[w]x%[h] ^
       ( xc:none -fill white -draw "roundrectangle 0,0 %[fx:w-1],%[fx:h-1] 20,20" ) ^
       -compose dstin -composite result.png

That uses the input image dimensions to make a sort of reverse mask with rounded corners, then uses the compose method "dstin" to composite that mask onto the input image to retain the shape of the mask.
